### PR TITLE
Consolidate kafka consumer configs

### PR DESCRIPTION
**Pull request by surekhasaharan**
Keep kafka consumer config properties in a common class for easy maintenance and avoid bugs due to missing properties.

### CODE REVIEW

File: extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaIndexTask.java [hunk 1]
```diff
@@ -96,20 +96,16 @@ long getPollRetryMs()
     try {
       Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
 
+      final Map<String, Object> consumerConfigs = KafkaConsumerConfigs.getConsumerProperties();
       final Properties props = new Properties();
+      props.putAll(consumerConfigs);
```
Comment (jon-wei) [inline extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaIndexTask.java]: This should be called after `KafkaRecordSupplier.addConsumerPropertiesFromConfig`

File: extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java [hunk 2]
```diff
@@ -193,15 +191,12 @@ public static void addConsumerPropertiesFromConfig(
 
   private KafkaConsumer<byte[], byte[]> getKafkaConsumer()
   {
+    final Map<String, Object> consumerConfigs = KafkaConsumerConfigs.getConsumerProperties();
     final Properties props = new Properties();
-
-    props.setProperty("metadata.max.age.ms", "10000");
-    props.setProperty("group.id", StringUtils.format("kafka-supervisor-%s", RandomIdUtils.getRandomId()));
+    props.putAll(consumerConfigs);
```
Comment (jon-wei) [inline extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java]: This should be called after KafkaRecordSupplier.addConsumerPropertiesFromConfig

File: extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaIndexTask.java (hunk 1 above)
Comment (surekhasaharan) [reply extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaIndexTask.java]: ah ok, yeah i was wondering if the order matters.

File: extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java (hunk 2 above)
Comment (surekhasaharan) [reply extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java]: ok

File: integration-tests/src/test/java/org/apache/druid/tests/indexer/ITKafkaIndexingServiceTest.java [hunk 3]
```diff
@@ -150,6 +150,8 @@ public void testKafka()
       LOG.info("supervisorSpec name: [%s]", INDEXER_FILE);
       Properties consumerProperties = new Properties();
       consumerProperties.put("bootstrap.servers", config.getKafkaInternalHost());
+      consumerProperties.put("auto.offset.reset", "none");
```
Comment (jihoonson) [inline integration-tests/src/test/java/org/apache/druid/tests/indexer/ITKafkaIndexingServiceTest.java]: Can we use `KafkaConsumerConfigs` here, too?

**Comment by jihoonson** [comment]
@surekhasaharan would you please check the integration test failure? It may be legit.

File: integration-tests/src/test/java/org/apache/druid/tests/indexer/ITKafkaIndexingServiceTest.java (hunk 3 above)
Comment (surekhasaharan) [reply integration-tests/src/test/java/org/apache/druid/tests/indexer/ITKafkaIndexingServiceTest.java]: yeah, will change

**Comment by surekhasaharan** [comment]
> @surekhasaharan would you please check the integration test failure? It may be legit.

Yes checking that.

**Comment by surekhasaharan** [comment]
> > @surekhasaharan would you please check the integration test failure? It may be legit.
> 
> Yes checking that.

I updated the zookeeper and kafka version in the setup.sh file for base docker image `imply/druiditbase`, which might be required since kafka supervisor uses client version `2.1.0`. The tests might still fail until we push the updated base image to docker repository.

**Pull request merged**